### PR TITLE
Revamp other transaction views

### PR DESCRIPTION
**Pull request by ThibautBremand**

### CODE REVIEW

File: packages/extension/cypress/e2e/trustlines.cy.ts [hunk 1]
```diff
@@ -287,12 +286,10 @@ const validateTrustlineTx = (destinationAddress: string, currency: string, limit
   cy.get('h1[data-testid="page-title"]').should('have.text', 'Set Trustline');
 
   // Should have the proper information
-  cy.contains('Issuer:').next().should('have.text', destinationAddress);
```
Comment (FlorianBouron) [inline packages/extension/cypress/e2e/trustlines.cy.ts:290]: Why is there less information now?

File: packages/extension/src/components/molecules/XRPLTransaction/XRPLTransaction.tsx [hunk 2]
```diff
@@ -195,16 +203,19 @@ export const XRPLTransaction: FC<XRPLTxProps> = ({
   const renderAmount = (params: {
     title: string;
     value: Amount;
+    showIssuer?: boolean;
```
Comment (FlorianBouron) [inline packages/extension/src/components/molecules/XRPLTransaction/XRPLTransaction.tsx:206]: I think showIssuer should be true by default.

File: packages/extension/src/components/pages/AddNewTrustline/AddNewTrustline.tsx [hunk 3]
```diff
@@ -285,77 +252,75 @@ export const AddNewTrustline: FC = () => {
...
-      issuer: params.limitAmount.issuer,
-      token: params.limitAmount.currency,
-      limit: Number(params.limitAmount.value),
+      issuer: params.transaction?.LimitAmount.issuer,
+      token: params.transaction?.LimitAmount.currency,
+      limit: Number(params.transaction.LimitAmount.value),
       noRipple
     };
-  }, [params.limitAmount, params.flags]);
+  }, [params.transaction]);
```
Comment (FlorianBouron) [inline packages/extension/src/components/pages/AddNewTrustline/AddNewTrustline.tsx:296]:
```suggestion
  }, [params.transaction?.LimitAmount]);
```

File: packages/extension/src/components/pages/SendPayment/ConfirmPayment/ConfirmPayment.tsx [hunk 4]
```diff
@@ -132,95 +165,20 @@ export const ConfirmPayment: FC<ConfirmPaymentProps> = ({
...
-      title="Confirm Payment"
-      onBackClick={onClickBack}
-      style={{
-        height: '100%',
-        display: 'flex',
-        flexDirection: 'column',
-        justifyContent: 'space-around'
-      }}
+    <TransactionPage
+      title="Send Payment"
```
Comment (FlorianBouron) [inline packages/extension/src/components/pages/SendPayment/ConfirmPayment/ConfirmPayment.tsx:170]:
```suggestion
      title="Submiy Payment"
```
Would this make more sense as the button is named: `Submit`?

File: packages/extension/src/components/pages/SendPayment/PreparePayment/PreparePayment.tsx [hunk 5]
```diff
@@ -51,7 +50,7 @@ export interface PreparePaymentProps {
     token: string;
     value: string;
     memos?: Memo[];
-    destinationTag?: string;
+    destinationTag?: number;
```
Comment (FlorianBouron) [inline packages/extension/src/components/pages/SendPayment/PreparePayment/PreparePayment.tsx:53]: Why did you change this type?

File: packages/extension/src/utils/format.test.ts [hunk 6]
```diff
@@ -156,6 +156,148 @@ describe('formatFlags', () => {
     };
     expect(formatFlags(flags, 'NFTokenMint')).toBe('Transferable');
   });
+
+  it('should format OfferCreate flags correctly when given as a number', () => {
+    const flags = 0x00010000 | 0x00020000; // both tfPassive and tfImmediateOrCancel flags are set
```
Comment (FlorianBouron) [inline packages/extension/src/utils/format.test.ts:161]: Really minor, but could be imported from xrpl.js

File: packages/extension/src/components/pages/SendPayment/ConfirmPayment/ConfirmPayment.tsx (hunk 4 above)
Comment (ThibautBremand) [reply packages/extension/src/components/pages/SendPayment/ConfirmPayment/ConfirmPayment.tsx:170]: I don't know!

File: packages/extension/cypress/e2e/trustlines.cy.ts (hunk 1 above)
Comment (ThibautBremand) [reply packages/extension/cypress/e2e/trustlines.cy.ts:290]: Because I did not find a nice way to display the issuer on the view. It looked bad imo. But I can put it back, I just need to revert my latest commit

File: packages/extension/src/components/pages/Transaction/Transaction.tsx [hunk 7]
```diff
@@ -1,82 +1,60 @@
...
+import { parseAmount, parsePaymentFlags } from '../../../utils';
+import { parseBaseParamsFromURLParamsNew } from '../../../utils/baseParams';
 import { serializeError } from '../../../utils/errors';
-import { BaseTransaction } from '../../organisms/BaseTransaction/BaseTransaction';
-import { AsyncTransaction, PageWithTitle } from '../../templates';
+import { TransactionDetails } from '../../organisms';
+import { AsyncTransaction, TransactionPage } from '../../templates';
 
-interface Params extends BaseTransactionParams {
+interface Params {
```
[the remark this answers is not in the document]
Comment (ThibautBremand) [reply packages/extension/src/components/pages/Transaction/Transaction.tsx:30]: Like all the other views, we only need a transaction object now

File: packages/extension/src/utils/format.test.ts (hunk 6 above)
Comment (ThibautBremand) [reply packages/extension/src/utils/format.test.ts:161]: It's from create offer, it's a bad rebase

File: packages/extension/src/components/pages/Transaction/Transaction.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (FlorianBouron) [reply packages/extension/src/components/pages/Transaction/Transaction.tsx:30]: Correct, thanks :)

File: packages/extension/cypress/e2e/trustlines.cy.ts (hunk 1 above)
Comment (FlorianBouron) [reply packages/extension/cypress/e2e/trustlines.cy.ts:290]: I would let it for know, maybe we can think about it in another MR, I think it's important to keep it, though the layout will need an update to have it displayed the right way in the future.
Let's keep it for now :)

File: packages/extension/src/utils/format.test.ts (hunk 6 above)
Comment (ThibautBremand) [reply packages/extension/src/utils/format.test.ts:161]: oh I see it's the test file, ok i will change it

File: packages/extension/src/components/pages/AddNewTrustline/AddNewTrustline.tsx (hunk 3 above)
Comment (ThibautBremand) [reply packages/extension/src/components/pages/AddNewTrustline/AddNewTrustline.tsx:296]: then it should be
[params.transaction?.Flags, params.transaction?.LimitAmount]

**Review by FlorianBouron — APPROVED** [review]
